### ISSUE TITLE
UMAP broken

### ISSUE THREAD

**Issue by pcm32** [ebi-gene-expression-group/scanpy-scripts#58]
Running on the latest container (quay.io/biocontainers/scanpy-scripts:0.2.4.post4--py_2) like this:

```
PYTHONIOENCODING=utf-8 scanpy-run-umap --use-graph 'neighbors' --key-added '' --export-embedding embeddings.csv  --input-format 'anndata' input.h5   --show-obj stdout --output-format 'anndata' output.h5
```

we get the following error:

```python
Traceback (most recent call last):
  File "/usr/local/bin/scanpy-run-umap", line 10, in <module>
    sys.exit(UMAP_CMD())
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/scanpy_scripts/cmd_utils.py", line 39, in cmd
    adata = _read_obj(input_obj, input_format=input_format)
  File "/usr/local/lib/python3.6/site-packages/scanpy_scripts/cmd_utils.py", line 71, in _read_obj
    adata = sc.read(input_obj)
  File "/usr/local/lib/python3.6/site-packages/scanpy/readwrite.py", line 79, in read
    backup_url=backup_url, cache=cache, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/scanpy/readwrite.py", line 439, in _read
    return read_h5ad(filename, backed=backed)
  File "/usr/local/lib/python3.6/site-packages/anndata/readwrite/read.py", line 446, in read_h5ad
    return AnnData(*_read_args_from_h5ad(filename=filename, chunk_size=chunk_size))
  File "/usr/local/lib/python3.6/site-packages/anndata/readwrite/read.py", line 480, in _read_args_from_h5ad
    _read_key_value_from_h5(f, d, key, chunk_size=chunk_size)
  File "/usr/local/lib/python3.6/site-packages/anndata/readwrite/read.py", line 504, in _read_key_value_from_h5
    _read_key_value_from_h5(f, d[key_write], key + '/' + k, k, chunk_size)
  File "/usr/local/lib/python3.6/site-packages/anndata/readwrite/read.py", line 504, in _read_key_value_from_h5
    _read_key_value_from_h5(f, d[key_write], key + '/' + k, k, chunk_size)
  File "/usr/local/lib/python3.6/site-packages/anndata/readwrite/read.py", line 538, in _read_key_value_from_h5
    key, value = postprocess_reading(key, value)
  File "/usr/local/lib/python3.6/site-packages/anndata/readwrite/read.py", line 535, in postprocess_reading
    value = value.astype(new_dtype)
ValueError: invalid shape in fixed-type tuple.
```

**Comment by nh3** [comment]
This seems not related to the umap step per se, but the input object was not correctly read in. Is there anything peculiar with this input object?

**Comment by pcm32** [comment]
It is a dataset coming from HCA... I'll share the history for your perusal. Wendi from training also reported the same issue though some days ago, but I don't know what dataset she had. I think that the atlas based datasets tend to work well.

**Comment by pcm32** [comment]
Inside the container, the versions currently used are:

```
Click           7.0
anndata         0.6.19
scanpy          1.4.3
scanpy-scripts  0.2.4.post4
```

**Comment by pcm32** [comment]
This has been fixed by downgrading h5py to 2.9.0
